File: frontend/src/dtos/models_dtos/request_dto/create_project_dto.ts
import type { ProjectPosition } from "../../enums/ProjectPosition"
import type { ProjectStatus } from "../../enums/ProjectStatus"
import type { TechnologyDTO } from "../technology_dto"


export interface CreateProjectDTO {
    
    name: string
    githubUrl: string
    status: ProjectStatus
    userPosition: ProjectPosition
    technologies: TechnologyDTO[]
}

File: backend/src/main/java/com/grouply/backend/project/Dtos/CreateProjectDTO.java
package com.grouply.backend.project.Dtos;

import com.grouply.backend.project.ProjectStatus;
import com.grouply.backend.project_member.ProjectPosition;
//import com.grouply.backend.technology.Technology;

import com.grouply.backend.technology.dto.TechnologyDTO;
import lombok.Data;


import java.time.LocalDateTime;

import java.util.List;
import java.util.Set;

@Data
public class CreateProjectDTO {

    private String name;
    private ProjectStatus status;
    private Set<TechnologyDTO> technologies;
    private List<String> defaultDmLinks;

    // this is for user to select his position its not actually related to the project
    private ProjectPosition userPosition;

}
